### PR TITLE
fix(kpop): bind aria-controls to trigger [KHCP-12832]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Add `aria-label` to dropdown trigger button

### CODE REVIEW

File: src/components/KDropdown/KDropdown.vue [hunk 1]
```diff
@@ -29,6 +29,7 @@
             <KButton
               v-if="triggerButtonText"
               :appearance="appearance"
+              :aria-label="isToggled.value ? 'Close dropdown' : 'Open dropdown'"
```
Comment (adamdehaven) [inline src/components/KDropdown/KDropdown.vue:32]: I'm not 100% sure this should be hard-coded as the Kongponent has no idea as to the context in which it's being used (e.g. there may be far better text here the app would be able to provide on its own).

e.g. if the dropdown is to "pick a region" and already has a label or other contextual text around the dropdown, adding _more_ labels would actually be harmful to the experience

Comment (portikM) [reply src/components/KDropdown/KDropdown.vue]: I think you're right here. Decided to go with a different approach

File: src/components/KPop/KPop.vue [hunk 2]
```diff
@@ -287,11 +287,15 @@ onMounted(() => {
     // we don't set any other click event listeners on purpose to avoid conflict of event listeners
     document?.addEventListener('click', clickHandler)
 
-    if (popoverTrigger.value && props.trigger === 'hover') {
-      popoverTrigger.value.addEventListener('mouseenter', showPopover)
-      popoverTrigger.value.addEventListener('focus', showPopover)
-      popoverTrigger.value.addEventListener('mouseleave', hidePopover)
-      popoverTrigger.value.addEventListener('blur', hidePopover)
+    if (popoverTrigger.value) {
+      popoverTrigger.value.setAttribute('aria-controls', popoverId)
```
Comment (adamdehaven) [inline src/components/KPop/KPop.vue:291]: what if the trigger already has this attribute set?

Comment (portikM) [reply src/components/KPop/KPop.vue]: Changed the logic a bit (to make sure to bind it to the button and not the wrapper div)

File: src/components/KDropdown/KDropdown.vue [hunk 3]
```diff
@@ -30,6 +30,7 @@
               v-if="triggerButtonText"
               :appearance="appearance"
               class="dropdown-trigger-button"
+              data-dropdown-trigger="true"
```
Comment (adamdehaven) [inline src/components/KDropdown/KDropdown.vue:33]: if this isn't documented we're likely not going to know what it's for 😬

Comment (portikM) [reply src/components/KDropdown/KDropdown.vue:33]: Added a comment in KPop (not where the attribute is defined but rather where it's being used - I'm guessing someone who wants to find out what it's for will try searching for references in the project)

**Pull request merged**